app/javascript/chat/ChatChannelSettings/ChatChannelSettings.jsx

```diff
@@ -305,6 +314,54 @@ export default class ChatChannelSettings extends Component {
     }
   };
 
+  toggleScreens = () => {
+    const { displaySettings, displayMembershipManager } = this.state;
+
+    this.setState({
+      displaySettings: !displaySettings,
+      displayMembershipManager: !displayMembershipManager,
+    });
+  };
+
+  handleUpdateMembershipRole = async (e) => {
+    const { membershipId, role } = e.target.dataset;
+    const { chatChannel } = this.state;
+    const response = await updateMembershipRole(
+      membershipId,
+      chatChannel.id,
+      role,
+    );
+    const { message } = response;
+    if (response.success) {
+      this.componentDidMount();
+      this.setState((prevState) => {
+        const { activeMemberships } = prevState;
+        const updatedActiveMemberships = activeMemberships.map(
+          (activeMembership) => {
+            if (activeMembership.membership_id === Number(membershipId)) {
+              // eslint-disable-next-line no-param-reassign
+              activeMembership.role = role;
```

Comment:
To avoid mutating the existing memberships, you can do the following:

```suggestion
              return { ...activeMembership, role };
```